Mage/src/main/java/mage/filter/predicate/other/DamagedPlayerThisTurnPredicate.java

```diff
@@ -29,13 +36,19 @@ public boolean apply(ObjectSourcePlayer<Controllable> input, Game game) {
             case YOU:
                 PlayerDamagedBySourceWatcher watcher = game.getState().getWatcher(PlayerDamagedBySourceWatcher.class, playerId);
                 if (watcher != null) {
+                    if (combatDamageOnly) {
```

Comment:
I don't love duplicating this logic in four places, might work better to refactor the watcher method to take the combatDamageOnly parameter.